include/openmc/particle_data.h

```diff
@@ -0,0 +1,453 @@
+                            //!< current tallies
+  Position r_last_;         //!< previous coordinates
+  Direction u_last_;        //!< previous direction coordinates
+  double wgt_last_ {1.0};   //!< pre-collision particle weight
+  double wgt_absorb_ {0.0}; //!< weight absorbed for survival biasing
+
+  // What event took place
+  bool fission_ {false};  //!< did particle cause implicit fission
+  TallyEvent event_;      //!< scatter, absorption
+  int event_nuclide_;     //!< index in nuclides array
+  int event_mt_;          //!< reaction MT
+  int delayed_group_ {0}; //!< delayed group
+
+  // Post-collision physical data
+  int n_bank_ {0};        //!< number of fission sites banked
+  int n_bank_second_ {0}; //!< number of secondary particles banked
+  double wgt_bank_ {0.0}; //!< weight of fission sites banked
+  int n_delayed_bank_[MAX_DELAYED_GROUPS]; //!< number of delayed fission
+                                           //!< sites banked
+
+  // Indices for various arrays
+  int surface_ {0};        //!< index for surface particle is on
+  int cell_born_ {-1};     //!< index for cell particle was born in
+  int material_ {-1};      //!< index for current material
+  int material_last_ {-1}; //!< index for last material
+
+  // Boundary information
+  BoundaryInfo boundary_;
+
+  // Temperature of current cell
+  double sqrtkT_ {-1.0};     //!< sqrt(k_Boltzmann * temperature) in eV
+  double sqrtkT_last_ {0.0}; //!< last temperature
+
+  // Statistical data
+  int n_collision_ {0}; //!< number of collisions
+
+  // Track output
+  bool write_track_ {false};
+
+  // Current PRNG state
+  uint64_t seeds_[N_STREAMS]; // current seeds
+  int stream_;                // current RNG stream
+
+  // Secondary particle bank
+  vector<ParticleBank> secondary_bank_;
+
+  int64_t current_work_; // current work index
+
+  vector<double> flux_derivs_; // for derivatives for this particle
+
+  vector<FilterMatch> filter_matches_; // tally filter matches
+
+  vector<std::vector<Position>> tracks_; // tracks for outputting to file
+
+  vector<NuBank> nu_bank_; // bank of most recently fissioned particles
+
+  // Global tally accumulators
+  double keff_tally_absorption_ {0.0};
+  double keff_tally_collision_ {0.0};
+  double keff_tally_tracklength_ {0.0};
+  double keff_tally_leakage_ {0.0};
+
+  bool trace_ {false}; //!< flag to show debug information
+
+  double collision_distance_; // distance to particle's next closest collision
+
+  int n_event_ {0}; // number of events executed in this particle's history
+
+// DagMC state variables
+#ifdef DAGMC
+  moab::DagMC::RayHistory history_;
+  Direction last_dir_;
+#endif
+
+  int64_t n_progeny_ {0}; // Number of progeny produced by this particle
+
+public:
+  //==========================================================================
+  // Methods and accessors
+
+  NuclideMicroXS& neutron_xs(const int& i) { return neutron_xs_[i]; }
```

Comment:
I don't see any particular reason to pass a const reference instead of just passing by value, which is preferred for simple scalar types like `int` and `double`. Can you change this (and others) to be just `int i`?

Reply:
Ah, good point. CPP core guidelines agree with you on this. Moreover, they claim there is overhead in passing by const reference. I find that interesting, because I was told over the past summer that the compiler will choose to pass by value if it deems that's more efficient when we're doing pass by reference and don't end up modifying the parameter.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in

Reply:
OK, latest commit incorporates your suggestion